storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -139,12 +139,12 @@
                   (.add local pair) 
 
                   ;;Using java objects directly to avoid performance issues in java code
-                  (let [node+port (get @task->node+port task)]
-                    (when (not (.get remoteMap node+port))
-                      (.put remoteMap node+port (ArrayList.)))
-                    (let [remote (.get remoteMap node+port)]
+                  (let []
+                    (when (not (.get remoteMap task))
+                      (.put remoteMap task (ArrayList.)))
+                    (let [remote (.get remoteMap task)]
                       (.add remote (TaskMessage. task (.serialize serializer tuple)))
-                     )))) 
+                    ))))
```

Comment:
@d2r 
Yes, TransferDrainer doesn't have to do it.
Btw, it makes me think that we may don't need to play with TransferDrainer since its main role is grouping tuples by node+port.

Could you confirm that I'm right?